crates/trie/src/prefix_set/mod.rs

```diff
@@ -121,6 +121,9 @@ impl PrefixSetMut {
             self.keys.dedup();
         }
 
+        // we need to shrink in both the sorted and non-sorted cases because deduping may have
+        // occurred either on `freeze`, or during `contains`.
+        self.keys.shrink_to_fit();
```

Comment:
This is not needed if `Arc<[u8]>` is used in `PrefixSet` instead. I would lean into this since shrink_to_fit might create another allocation.

Reply:
hmm, how can i perform this conversion without also allocating? the `From<Vec<T>> for Arc<[T]>` conversion allocates, it seems like the slice alloc does too

https://github.com/rust-lang/rust/blob/47ecded3525392b77843534bed69b4302f9af8d2/library/alloc/src/sync.rs#L3416-L3444

Reply:
bump @DaniPopes